### PR TITLE
[ENH] create demo for event related experiment

### PR DESCRIPTION
**Pull request by Remi-Gau**
- [x] add function to create default model.json
- [x] create demo to download face SPM repetition dataset, convert it to BIDS and run the analysis on it

### CODE REVIEW

File: demos/spm_face_rep/face_rep_run.m [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+%%
+checkDependencies();
+
+%% Run batches
+% reportBIDS(opt);
+% bidsCopyRawFolder(opt, 1);
+%
+% bidsSTC(opt);
+%
+% bidsSpatialPrepro(opt);
```
Comment (Remi-Gau) [inline demos/spm_face_rep/face_rep_run.m:50]: uncomment those lines

File: demos/spm_face_rep/face_rep_run.m [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+% The following do not run on octave for now (because of spmup)
+% anatomicalQA(opt);
+% bidsResliceTpmToFunc(opt);
+
+% DEBUG
```
Comment (Remi-Gau) [inline demos/spm_face_rep/face_rep_run.m:56]:
```suggestion
% DEBUG spmup crashes: investigate
```

**Review by Remi-Gau — COMMENTED** [review]
fix a few things

**Comment by Remi-Gau** [comment]
@marcobarilari @fedefalag 

have a look at this when you have time.

You can run the new face repetition demo in the `demos` folder. Nothing super new for that one except maybe a function to show how to do some BIDS conversion with matlab.

The new helpful feature are 2 functions that help create the model.json which can be a bit complex to write down.

- `createEmptyModel` creates the content of JSON file but does not save an actual file.
- `createDefaultModel` calls `createEmptyModel` to get some basic content and then checks all the events.tsv files a BIDS dataset to create some default design matrix and autocontrast in the json file for all the conditions that present in this dataset.

Have a look at the code and the help sections to make sure they are "good enough".

I think it can be good to "checkout" the branch of that pull request and try to run some of that on your computer directly.

File: demos/spm_face_rep/face_rep_run.m [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+% functionalQA(opt);
+
+bidsSmoothing(FWHM, opt);
+
+% The following crash on Travis CI
+bidsFFX('specifyAndEstimate', opt, FWHM);
+bidsFFX('contrasts', opt, FWHM);
+
+%TODO
+bidsResults(opt, FWHM);
```
Comment (CerenB) [inline demos/spm_face_rep/face_rep_run.m:73]: this one is crashing but it is not really important. I might need to set things up right in `getOptions.m` probably. 
```
WORKFLOW: COMPUTING GLM RESULTS

Undefined function or variable 'batchName'.

Error in bidsResults (line 83)
  saveAndRunWorkflow(matlabbatch, batchName, opt);

Error in face_rep_run (line 73)
bidsResults(opt, FWHM);
```

**Review by CerenB — COMMENTED** [review]
very good demo!

**Pull request merged**